### ISSUE TITLE
Post-training dataset curation using instruction-level selection

### ISSUE THREAD

**Issue by Ali-Elganzory** [OpenEuroLLM/Taskboard#416]
__Goal__
Curate a high quality post-training dataset by performing instruction-level selection. The selection will be based onmultiple quality dimensions measured/annotated using Propella.

__Description__
As a first step, we ran SFT on 6 dataset sources in (#330). The goal was to have a baseline. Next, all of the instructions across all of the 6 dataset sources got annotated with multiple quality dimensions using Propella. All of the annotated 6 dataset sources are merged into a single pool. From which, a dataset is curated to contain only the "excellent" instructions from the merged pool. Further details about the curation are in (#318).

**Comment by Ali-Elganzory** [comment]
## Update: Performed SFT on the curated dataset ("excellent" instructions)
- 2 training runs using 2 different pre-trained models:
  - Olmo3: `allenai/Olmo-3-1025-7B`
  - OpenEuroLLM Prelude: `birgermoell/oellm-9b-128k-theta32m-prelude-anneal300b`
- Budget: 2 epochs
  - Full curated-v4 for 1 epoch in tokens
  - Using Olmo3 tokenizer: 25,947,239,071 (~26B) tokens
  - Using Prelude tokenizer: 29,154,931,289 (~29B) tokens

## Training loss and accuracy
- The curves look healthy, no abnormal signs
- Prelude seems to be leading olmo on both token accuracy and loss over time
- Both have different training lengths because they use different tokenizers
- W&B: https://wandb.ai/openeurollm-project/instruction-selection/workspace

<img width="1033" height="306" alt="Image" src="https://github.com/user-attachments/assets/2986f9eb-6ea8-4e44-97e0-548105a8e441" />

## Evaluation
### TODO
